flink-table-store-core/src/main/java/org/apache/flink/table/store/file/data/DataFileWriter.java

```diff
@@ -237,7 +210,14 @@ public RollingKvWriter(Supplier<KvFileWriter> writerFactory, long targetFileSize
     private Supplier<KvFileWriter> createWriterFactory(int level) {
         return () -> {
             try {
-                return new KvFileWriter(new KvBulkWriterFactory(), pathFactory.newPath(), level);
+                FileWriter.Factory<KeyValue, Metric> factory =
```

Comment:
Factory can be init in the class constructor?

Reply:
Yes, I agree.  It will be more clear if moving the factory initialization to the constructor.

Reply:
I decided to revert this change ( I mean putting the factory initialization into the constructor) , because it break the unit test ( I think the cause is the reuse of `KeyValueSerializer`).

After applying this changes : https://github.com/apache/flink-table-store/pull/117/commits/0dd583045be08cc19990c4c98370bab1aad3ddde,   the [unit test ](https://github.com/apache/flink-table-store/pull/117#issuecomment-1129526584)  works fine.